### ISSUE TITLE
[java] New rule: LongLiteralEndingWithLowercaseL

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#7027]
**Proposed Rule Name:** LongLiteralEndingWithLowercaseL

**Proposed Category:** Error Prone

**Description:**

To avoid confusion between 1 and l, use `L` to indicate long literals. The proposed rule name is aligned with IntelliJ

https://www.jetbrains.com/help/inspectopedia/LongLiteralEndingWithLowercaseL.html

**Code Sample:** 

```java
class Longs {
  long bad = 1l; // bad, may be confusing with some fonts
  long good = 1L; // good
}

```

**Possible Properties:**

none

--- 

Currently possible as XPath rule using `//NumericLiteral[matches(@Image,'.*l')]`
